src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -471,6 +471,31 @@ public static IEnumerable<object[]> GetAsync_IPBasedUri_Success_MemberData()
             }
         }
 
+        [Fact]
+        public async Task GetAsync_IPv6AddressInHostHeader_CorrectlyFormatted()
+        {
+            string host = "[::1234]:8080";
```

Comment:
Can you make the test a theory and validate with and without a port?

Reply:
Sure.